core/internal/testutils/testutils.go

```diff
@@ -450,3 +452,16 @@ func MustDecodeBase64(s string) (b []byte) {
 	}
 	return
 }
+
+// GetFreePort returns a free port.
+// NOTE: This approach is technically incorrect because the returned port
+// can still be taken by the time the caller attempts to bind to it.
+// Unfortunately, we can't specify zero port in P2P.V2.ListenAddresses at the moment.
```

Comment:
FYI I asked Research about this here: https://chainlink-core.slack.com/archives/C042C3WPDNH/p1691442588046129